lib/dal/src/audit_logging.rs

```diff
@@ -2,53 +2,53 @@
 
 use audit_logs::AuditLogsError;
 use audit_logs::AuditLogsStream;
-use chrono::Utc;
+use futures::StreamExt;
 use pending_events::PendingEventsError;
 use pending_events::PendingEventsStream;
 use shuttle_server::Shuttle;
 use shuttle_server::ShuttleError;
+use si_data_nats::async_nats;
+use si_data_nats::async_nats::jetstream::consumer::pull::BatchErrorKind;
+use si_data_nats::async_nats::jetstream::stream::ConsumerErrorKind;
 use si_events::audit_log::AuditLog;
 use si_events::audit_log::AuditLogKind;
+use si_events::Actor;
+use si_frontend_types::AuditLog as FrontendAuditLog;
 use telemetry::prelude::*;
 use thiserror::Error;
 use tokio_util::task::TaskTracker;
-use ulid::MonotonicError;
 
 use crate::DalContext;
 use crate::TenancyError;
-use crate::{ChangeSetError, ChangeSetId, TransactionsError, WorkspaceError, WorkspacePk};
-
-mod fake_data_for_frontend;
-
-pub use fake_data_for_frontend::filter_and_paginate;
-pub use fake_data_for_frontend::generate;
+use crate::TransactionsError;
+use crate::User;
+use crate::UserError;
+use crate::UserPk;
 
 #[remain::sorted]
 #[derive(Debug, Error)]
 pub enum AuditLoggingError {
+    #[error("async nats batch error: {0}")]
+    AsyncNatsBatch(#[from] async_nats::error::Error<BatchErrorKind>),
+    #[error("async nats consumer error: {0}")]
+    AsyncNatsConsumer(#[from] async_nats::error::Error<ConsumerErrorKind>),
     #[error("audit logs error: {0}")]
     AuditLogs(#[from] AuditLogsError),
-    #[error("change set error: {0}")]
-    ChangeSet(#[from] Box<ChangeSetError>),
-    #[error("change set not found: {0}")]
-    ChangeSetNotFound(ChangeSetId),
-    #[error("monotonic error: {0}")]
-    Monotonic(#[from] MonotonicError),
     #[error("pending events error: {0}")]
     PendingEventsError(#[from] PendingEventsError),
+    #[error("serde json error: {0}")]
+    SerdeJson(#[from] serde_json::Error),
     #[error("shuttle error: {0}")]
     Shuttle(#[from] ShuttleError),
     #[error("transactions error: {0}")]
     Transactions(#[from] Box<TransactionsError>),
-    #[error("ulid decode error: {0}")]
-    UlidDecode(#[from] ulid::DecodeError),
-    #[error("workspace error: {0}")]
-    Workspace(#[from] Box<WorkspaceError>),
-    #[error("workspace not found: {0}")]
-    WorkspaceNotFound(WorkspacePk),
+    #[error("user error: {0}")]
+    User(#[from] Box<UserError>),
+    #[error("user not found for id: {0}")]
+    UserNotFound(UserPk),
 }
 
-pub type AuditLoggingResult<T> = Result<T, AuditLoggingError>;
+type Result<T> = std::result::Result<T, AuditLoggingError>;
```

Comment:
Do you like this pattern? I think I am starting to like this myself. It's easy to work with within modules and it almost never needs to leak outside. And if it does, you can always pub-export an `AuditLoggingResult` type alias. Anyway, cool!

Reply:
I do 😄  because I don't actually use the type externally: it is for convenience and footgun removal. I'd argue we should move towards this pattern because exporting result types feels like the wrong move "by default".

Reply:
I am a fan. A lot of crates do this, too!